src/poetry/core/version/markers.py

```diff
@@ -682,39 +635,20 @@ def append(self, marker: BaseMarker) -> None:
         self._markers.append(marker)
 
     def intersect(self, other: BaseMarker) -> BaseMarker:
-        if other.is_any():
-            return self
-
-        if other.is_empty():
-            return other
-
-        new_markers = []
-        if isinstance(other, (SingleMarker, MultiMarker)):
-            for marker in self._markers:
-                intersection = marker.intersect(other)
-
-                if not intersection.is_empty():
-                    new_markers.append(intersection)
-        elif isinstance(other, MarkerUnion):
-            for our_marker in self._markers:
-                for their_marker in other.markers:
-                    intersection = our_marker.intersect(their_marker)
-
-                    if not intersection.is_empty():
-                        new_markers.append(intersection)
+        if isinstance(other, MultiMarker):
+            return other.intersect(self)
```

Comment:
Same here. It seems it just changes nothing but the order (and IMO without it, it's even better).